courses/models_test.py

```diff
@@ -15,10 +24,248 @@ def test_to_string(self):
         assert "{}".format(prog) == "Title"
 
 
-class CourseTests(TestCase):
+class CourseModelsMixin(TestCase):
+    """Mixin for Course models"""
+
+    @classmethod
+    def setUpClass(cls):
+        super(CourseModelsMixin, cls).setUpClass()
+        cls.course = CourseFactory.create(title="Title")
+
+    def setUp(self):
+        super(CourseModelsMixin, self).setUp()
+        self.now = datetime.now(pytz.utc)
+
+    def create_run(self, course=None, start=None, end=None, enr_start=None, enr_end=None):
+        """helper function to create course runs"""
+        # pylint: disable=too-many-arguments
+        return CourseRunFactory.create(
+            course=course or self.course,
+            title="Title Run",
+            start_date=start,
+            end_date=end,
+            enrollment_start=enr_start,
+            enrollment_end=enr_end,
+        )
+
+
+class CourseTests(CourseModelsMixin):
     """Tests for Course model"""
 
     def test_to_string(self):
         """Test for __str__ method"""
-        course = CourseFactory.build(title="Title")
-        assert "{}".format(course) == "Title"
+        assert "{}".format(self.course) == "Title"
+
+    def test_get_next_run_no_run(self):
+        """Test for the get_next_run method"""
+        assert self.course.get_next_run() is None
+
+    def test_get_next_run_only_past(self):
```

Comment:
I actually thought about doing this, but we will basically move the messiness from inside the test to the parameters..